src/devices/secplus_v2.c

```diff
@@ -0,0 +1,548 @@
+        if (bits.bb[0][0] & 0xB0) {
+            if (decoder->verbose)
+                fprintf(stderr, "%s: DECODE_FAIL_SANITY 0X00XXXX\n", __func__);
+            continue;
+        }
+
+        if (bits.bb[0][0] & 0xC0) {
+            if (decoder->verbose)
+                (void)fprintf(stderr, "%s: Set 2\n", __func__);
+            _decode_v2_half(&bits, rolling_2, &fixed_2, decoder->verbose);
+        }
+        else {
+            if (decoder->verbose)
+                (void)fprintf(stderr, "%s: Set 1\n", __func__);
+            _decode_v2_half(&bits, rolling_1, &fixed_1, decoder->verbose);
+        }
+
+        // break if we've recived both parts
+        if (fixed_1.bits_per_row[0] > 1 && fixed_2.bits_per_row[0] > 1) {
+            break;
+        }
+
+        // i++;
+    }
+
+    if (decoder->verbose)
+        fprintf(stderr, "%s M fixed_1.bits_per_row = %u\tfixed_2.bits_per_row = %u\n",
+                __func__, fixed_1.bits_per_row[0], fixed_2.bits_per_row[0]);
+
+    // Do was have what we need ??
+    if (fixed_1.bits_per_row[0] == 0 || fixed_2.bits_per_row[0] == 0) {
+        //  No?  Awww F'ck it then
+        if (decoder->verbose)
+            fprintf(stderr, "%s: DECODE_FAIL_SANITY\n", __func__);
+        return DECODE_FAIL_SANITY;
+    }
+
+    if (decoder->verbose > 1) {
+        bitrow_debugf(fixed_1.bb[0], fixed_1.bits_per_row[0], "%s: fixed bit 1 : ", __func__);
+        fprintf(stderr, "%s rolling_1 : ", __func__);
+        for (int i = 0; i < 9; i++) {
+            fprintf(stderr, "%d ", rolling_1[i]);
+        }
+        fprintf(stderr, "\n");
+
+        bitrow_debugf(fixed_2.bb[0], fixed_2.bits_per_row[0], "%s: fixed bit 2 : ", __func__);
+        fprintf(stderr, "%s rolling_2 : ", __func__);
+        for (int i = 0; i < 9; i++) {
+            fprintf(stderr, "%d ", rolling_2[i]);
+        }
+        fprintf(stderr, "\n\n\n");
+    }
+
+    // Assemble rolling_1[] and rolling_2[] into rolling_digits[]
+    uint8_t rolling_digits[24] = {0};
+    uint8_t *r;
+
+    r    = rolling_digits;
+    *r++ = rolling_2[8];
+    *r++ = rolling_1[8];
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    if (decoder->verbose > 1) {
+        fprintf(stderr, "rolling_digits :");
+        for (int i = 0; i < 18; i++) {
+            fprintf(stderr, "%d ", rolling_digits[i]);
+        }
+        fprintf(stderr, "\n\n");
+    }
+
+    // compute rolling_total from rolling_digits[]
+    uint32_t rolling_total = 0;
+    uint32_t rolling_temp  = 0;
+
+    for (int i = 0; i < 18; i++) {
+        rolling_temp = (rolling_temp * 3) + rolling_digits[i];
+        // fprintf(stderr, ">> %12d\t%d\n", rolling_temp, rolling_digits[i]);
+    }
+
+    rolling_total = reverse32(rolling_temp);
```

Comment:
```suggestion
    rolling_total = reverse32(rolling_temp) >> 4;
```

Reply:
This is needed because the rolling code is a 28-bit integer, so it must be shifted right by four bits after going through a 32-bit reversal.

Reply:
that *Was* in there,  I guess It got deleted at some point, thanks...